src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -42,7 +42,7 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
 
             // Authorize the request
             SecretManager secretManager = (SecretManager)controllerContext.Configuration.DependencyResolver.GetService(typeof(SecretManager));
-            HttpBindingMetadata httpFunctionMetadata = (HttpBindingMetadata)function.Metadata.InputBindings.FirstOrDefault(p => p.Type == BindingType.HttpTrigger);
+            HttpTriggerBindingMetadata httpFunctionMetadata = (HttpTriggerBindingMetadata)function.Metadata.InputBindings.FirstOrDefault(p => p.Type == BindingType.HttpTrigger);
```

Comment:
Currently, yes. The http out binding for http triggered functions should really be optional in the metadata file - we should just default one if not present.